### DISCUSSION TITLE
Convert script from a language to another thanks to Antlr

### DISCUSSION THREAD

**Discussion by NicoLaval** [antlr/antlr4#4068] [General]
Hi,

I wonder if it's possible with antlr to convert a script from a language A to a language B.

Imagine A & B cover the same use cases, do you think it's possible to describe them with EBNF files and make a bridge between A & B scripts?

**Comment by ericvergnaud** [comment]
Hi

Antlr is certainly a good tool, part of the solution, but it won't address fundamental typing model differences between languages.
As an example, it would be pretty challenging to convert code from a language that supports OOP (Java, CSharp, Python, TypeScript...) to a non OOP one (C, Rust...), or to convert code from a language that supports dynamic typing (Python, JavaScript) to one that requires static typing (C++, Java, C#...).
These conversions are far from trivial. They require parsing, but it's not the parser's job to perform the conversion.

**Comment by KvanTTT** [comment]
The issue should be moved to Discussions.
